Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -415,56 +415,63 @@ private extension _ChatMessage {
-                .map(_ChatMessage.init)
-        }
         localState = dto.localMessageState
-        
         isFlaggedByCurrentUser = dto.flaggedBy != nil
-
-        if dto.reactions.isEmpty {
-            latestReactions = []
-        } else {
-            latestReactions = Set(
-                MessageReactionDTO
-                    .loadLatestReactions(for: dto.id, limit: 5, context: context)
-                    .map { $0.asModel() }
-            )
-        }
+        quotedMessageId = dto.quotedMessage.map(\.id)
         
         if let currentUser = context.currentUser() {
+            isSentByCurrentUser = currentUser.user.id == dto.user.id
+            
             if dto.reactions.isEmpty {
-                currentUserReactions = []
+                $_currentUserReactions = { [] }
             } else {
-                currentUserReactions = Set(
-                    MessageReactionDTO
-                        .loadReactions(for: dto.id, authoredBy: currentUser.user.id, context: context)
-                        .map { $0.asModel() }
-                )
+                $_currentUserReactions = {
+                    Set(
+                        MessageReactionDTO
+                            .loadReactions(for: dto.id, authoredBy: currentUser.user.id, context: context)
+                            .map { $0.asModel() }
+                    )
+                }
             }
-            isSentByCurrentUser = currentUser.user.id == dto.user.id
         } else {
-            currentUserReactions = []
             isSentByCurrentUser = false
+            $_currentUserReactions = { [] }
+        }
+        
+        $_mentionedUsers = { Set(dto.mentionedUsers.map { $0.asModel() }) }
+        $_author = { dto.user.asModel() }
+        $_attachments = {
+            dto.attachments
+                .map { $0.asModel() }
+                .sorted {
+                    let index1 = $0.id?.index ?? Int.max
+                    let index2 = $1.id?.index ?? Int.max
+                    return index1 < index2
+                }
         }
         
-        attachments = dto.attachments
-            .map { $0.asModel() }
-            .sorted {
-                let index1 = $0.id?.index ?? Int.max
-                let index2 = $1.id?.index ?? Int.max
-                return index1 < index2
+        if dto.replies.isEmpty {
+            $_latestReplies = { [] }
+        } else {
+            $_latestReplies = {
+                MessageDTO
+                    .loadReplies(for: dto.id, limit: 5, context: context)
+                    .map(_ChatMessage.init)
             }
-
-        quotedMessageId = dto.quotedMessage.map(\.id)
+        }
+        
+        if dto.reactions.isEmpty {
+            $_latestReactions = { [] }
+        } else {
+            $_latestReactions = {
+                Set(
+                    MessageReactionDTO
+                        .loadLatestReactions(for: dto.id, limit: 5, context: context)
```

Comment:
Hardcoded limit is okay? Same for line 459.

Reply:
Not sure, just moved the original code to make compiler happy. In `ChannelDTO` there is originally a TODO for that.